sklearn/decomposition/tests/test_truncated_svd.py

```diff
@@ -177,24 +186,27 @@ def test_singular_values():
 
     apca = TruncatedSVD(n_components=2, algorithm='arpack',
                         random_state=rng).fit(X)
-    rpca = TruncatedSVD(n_components=2, algorithm='arpack',
-                        random_state=rng).fit(X)
-    assert_array_almost_equal(apca.singular_values_, rpca.singular_values_, 12)
+    pca = TruncatedSVD(n_components=2, algorithm='randomized',
+                       random_state=rng).fit(X)
+    assert_allclose(apca.singular_values_, pca.singular_values_, rtol=1e-2)
 
     # Compare to the Frobenius norm
     X_apca = apca.transform(X)
-    X_rpca = rpca.transform(X)
-    assert_array_almost_equal(np.sum(apca.singular_values_**2.0),
-                              np.linalg.norm(X_apca, "fro")**2.0, 12)
-    assert_array_almost_equal(np.sum(rpca.singular_values_**2.0),
-                              np.linalg.norm(X_rpca, "fro")**2.0, 12)
+    X_pca = pca.transform(X)
```

Comment:
I would split this test. The check with the norm can be done for each solver and does not require to have a solver as a base to make comparison. I had something similar in PCA